erpnext/manufacturing/report/bom_stock_report/bom_stock_report.py

```diff
@@ -6,52 +6,74 @@
+
+	return columns, data
 
 def get_columns():
-    """return columns"""
-    columns = [
-        _("Item") + ":Link/Item:150",
-        _("Description") + "::500",
-        _("Required Qty") + ":Float:100",
-        _("In Stock Qty") + ":Float:100",
-        _("Enough Parts to Build") + ":Float:200",
-    ]
+	"""return columns"""
+	columns = [
+		_("Item") + ":Link/Item:150",
+		_("Description") + "::500",
+		_("Required Qty") + ":Float:100",
+		_("In Stock Qty") + ":Float:100",
+		_("Enough Parts to Build") + ":Float:200",
+	]
 
-    return columns
+	return columns
 
 def get_bom_stock(filters):
-    conditions = ""
-    bom = filters.get("bom")
-
-    if filters.get("warehouse"):
-        warehouse_details = frappe.db.get_value("Warehouse", filters.get("warehouse"), ["lft", "rgt"], as_dict=1)
-        if warehouse_details:
-            conditions += " and exists (select name from `tabWarehouse` wh \
-        	    where wh.lft >= %s and wh.rgt <= %s and ledger.warehouse = wh.name)" % (warehouse_details.lft,
-                warehouse_details.rgt)
-        else:
-            conditions += " and ledger.warehouse = '%s'" % frappe.db.escape(filters.get("warehouse"))
-
-    else:
-        conditions += ""
-
-    return frappe.db.sql("""
-    		SELECT
-    	        bom_item.item_code ,
-    	        bom_item.description ,
-    	        bom_item.qty,
-    	        sum(ledger.actual_qty) as actual_qty,
-    	        sum(FLOOR(ledger.actual_qty /bom_item.qty))as to_build
-            FROM
-    	        `tabBOM Item` AS bom_item
-    	        LEFT JOIN `tabBin` AS ledger
-    		    ON bom_item.item_code = ledger.item_code
-    		    %s
-            WHERE
-    	        bom_item.parent = '%s' and bom_item.parenttype='BOM'
-
-            GROUP BY bom_item.item_code""" % (conditions, bom))
+	conditions = ""
+	bom = filters.get("bom")
+
+	if filters.get("warehouse"):
+		warehouse_details = frappe.db.get_value("Warehouse", filters.get("warehouse"), ["lft", "rgt"], as_dict=1)
+		if warehouse_details:
+			conditions += " and exists (select name from `tabWarehouse` wh \
+				where wh.lft >= %s and wh.rgt <= %s and ledger.warehouse = wh.name)" % (warehouse_details.lft,
+				warehouse_details.rgt)
+		else:
+			conditions += " and ledger.warehouse = '%s'" % frappe.db.escape(filters.get("warehouse"))
+
+	else:
+		conditions += ""
+
+	return frappe.db.sql("""
+			SELECT
+				bom_item.item_code ,
+				bom_item.description ,
+				bom_item.qty,
+				sum(ledger.actual_qty) as actual_qty,
+				sum(FLOOR(ledger.actual_qty /bom_item.qty))as to_build
+			FROM
+				`tabBOM Item` AS bom_item
+				LEFT JOIN `tabBin` AS ledger
+				ON bom_item.item_code = ledger.item_code
+				%s
+			WHERE
+				bom_item.parent = '%s' and bom_item.parenttype='BOM'
+
+			GROUP BY bom_item.item_code""" % (conditions, bom))
+
+
+def get_exploded_bom_stock(filters):
+	root_bom_name = filters.get("bom")
+	root_bom = frappe.get_doc("BOM", root_bom_name)
+
+	bom_tree_list = root_bom.traverse_tree()
```

Comment:
No need to traverse tree and execute the same query for all the sub BOMs in a for loop. There is a separate child table "BOM Explosion Item" inside BOM, which has all the exploded raw materials. So, in the above query, you just need to change the table from `tabBOM Item` to `tabBOM Explosion Item` in case of exploded view.

Reply:
Pushed changes in PR https://github.com/frappe/erpnext/pull/12506.
I couldn't reopen https://github.com/frappe/erpnext/pull/12475 . Not sure how to do this.